airbyte-workers/src/main/java/io/airbyte/workers/temporal/CancellationHandler.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers.temporal;
+
+import io.airbyte.workers.WorkerException;
+import io.temporal.activity.Activity;
+import io.temporal.activity.ActivityExecutionContext;
+import io.temporal.client.ActivityCompletionException;
+
+public interface CancellationHandler {
+
+  void heartbeat(Runnable callback) throws WorkerException;
+
+  class TemporalCancellationHandler implements CancellationHandler {
+
+    final ActivityExecutionContext context;
+
+    public TemporalCancellationHandler() {
+      context = Activity.getExecutionContext();
+    }
+
+    @Override
+    public void heartbeat(Runnable callback) throws WorkerException {
+      try {
+        context.heartbeat(null);
```

Comment:
can you explain what this line does? is the idea that from outside the worker cancel gets called via the temporal client so that the next time this heart beat happens an exception will be thrown?

Reply:
Yeah exactly.

Reply:
can we add a comment explaining this? I would not remember the reason for this a week from now :P